crates/rolldown/src/stages/generate_stage/compute_cross_chunk_links.rs

```diff
@@ -490,11 +489,12 @@ impl GenerateStage<'_> {
 
         let mut processed_entry_exports = FxHashSet::default();
         if let Some(entry_module_idx) = chunk.entry_module_idx() {
+          let exported_chunk_symbols = &index_chunk_exported_symbols[chunk_id];
           // If this's an entry point, we need to make sure the entry modules' exports are not minified.
           let entry_module = &self.link_output.metas[entry_module_idx];
           entry_module.canonical_exports(false).for_each(|(name, export)| {
             let export_ref = self.link_output.symbol_db.canonical_ref_for(export.symbol_ref);
-            if !self.link_output.used_symbol_refs.contains(&export_ref) {
+            if !exported_chunk_symbols.contains_key(&export.symbol_ref) {
```

Comment:
A export symbol in the entry module is used does not mean it should be exported, e.g.

For a json entry module, even if only `default` is used, all top-level named exports will also be included. If we used `self.link_output.used_symbol_refs` to filter out symbol, it would include `hello` which is actually inlined